lib/fswatcher/fswatcher.go

```diff
@@ -21,43 +21,58 @@ import (
 
 type FsEvent struct {
 	path string
+	time time.Time
 }
 
 var Tempnamer scanner.TempNamer
 
 type FsEventsBatch map[string]*FsEvent
 
+type trackedDir struct {
+	count       int
+	childEvents FsEventsBatch
+}
+
 type FsWatcher struct {
 	folderPath            string
 	notifyModelChan       chan<- FsEventsBatch
 	fsEvents              FsEventsBatch
 	fsEventChan           <-chan notify.EventInfo
 	WatchingFs            bool
 	notifyDelay           time.Duration
+	slowNotifyDelay       time.Duration
 	notifyTimer           *time.Timer
 	notifyTimerNeedsReset bool
 	inProgress            map[string]struct{}
 	folderID              string
 	ignores               *ignore.Matcher
+	trackedDirs           map[string]*trackedDir
```

Comment:
a bit of explanation between the interaction of this versus fsEvents could be explained, as it's not immediately obvious.